### DISCUSSION TITLE
Can't use singel role in guard web and api

### DISCUSSION THREAD

**Discussion by rickyanwar** [spatie/laravel-permission#2452] [Q&A]
I have app , use web and api ,same function, used spatie for manage the role and permission, now role only working like this we need create both of guard
admin guard for web
admin guard for api

I want is
admin guard *

I need just create one but can use in web or api

It can?

**Comment by drbyte** [comment]
Yes. You can create a role for one guard, and tell the application to always find roles based on that guard (regardless of what guard the user logs in with). You have to make one change in your user model, as explained in my comment posted here: https://github.com/spatie/laravel-permission/discussions/2421#discussioncomment-5785479

**Reply by rickyanwar** [reply]
i try to make singel Role name DESA with guard api , I try not to give any permission on that role,
![image](https://github.com/spatie/laravel-permission/assets/36995966/dce4a058-a42d-4e65-b670-b87f7c93a010)
![image](https://github.com/spatie/laravel-permission/assets/36995966/830fac1e-6868-4b5c-a10c-432dae4b5816)

I try to test in web controller to testing on web
![image](https://github.com/spatie/laravel-permission/assets/36995966/8abd7443-2a9e-4fd8-b971-4b20d84306c4)

when accessing the web still can access the permission not working
